code/datums/ai/_ai_controller.dm

```diff
@@ -109,6 +113,8 @@ multiple modular subtrees with behaviors
 		set_ai_status(AI_STATUS_ON)
 
 	RegisterSignal(pawn, COMSIG_MOB_LOGIN, .proc/on_sentience_gained)
+	if(ai_traits & AI_CAN_HIT_DECK && ai_status == AI_STATUS_ON && isliving(pawn))
```

Comment:
CC @tralezab and @AMonkeyThatCodes, it seems whack to put this on the AI controller itself, which should be as generic as possible...

Reply:
yeah this is bloat. should not be on the controller.